src/dynamichmc_inference.jl

```diff
@@ -43,39 +42,11 @@ function dynamichmc_inference(prob::DiffEqBase.DEProblem, alg, likelihood, prior
                               ϵ=0.001, initial=Float64[], num_samples=1000,
                               kwargs...)
     P = DynamicHMCPosterior(alg, prob, likelihood, priors, kwargs)
```

Comment:
@tpapp should this not be returning a `Float64` type? Does that lead to problem with `FluxGradientLogDensity`, I am getting `LoadError: MethodError: no method matching Flux.Tracker.TrackedReal{Float64}(::Float64)`?

Reply:
When called, `P` should return a `<: Real` which is finite or `-Inf`. The wrappers (transform, AD) will take care of converting it to the `LogDensityProblems.ValueGradient` type.

Reply:
It looks like the problem is coming from the `problem_new_parameters` call which creates a new problem with the new parameters `a` as `a` is now of `::NamedTuple{(:a,),Tuple{Flux.Tracker.TrackedReal{Float64}}` type instead. @ChrisRackauckas have you come across this? (I remember there was some work on Flux and DiffEq compatibility)

Reply:
replace `problem_new_parameters` with `remake(prob,u0=...)` so that `u0` gets the right Tracked/Dual number type.